zarr/storage.py

```diff
@@ -781,12 +781,15 @@ def __setitem__(self, key, value):
 
     def __delitem__(self, key):
         path = os.path.join(self.path, key)
+        temp_path = path + '.partial'
         if os.path.isfile(path):
-            os.remove(path)
+            os.rename(path, temp_path)
+            os.remove(temp_path)
         elif os.path.isdir(path):
             # include support for deleting directories, even though strictly
             # speaking these do not exist as keys in the store
-            shutil.rmtree(path)
+            os.rename(path, temp_path)
```

Comment:
This rename will fail if the `.partial` file already exist.

Reply:
Same as above.

Reply:
Actually would add we don't create a directory named `.partial` at all currently.